packages/fxa-content-server/app/tests/spec/lib/metrics.js

```diff
@@ -1051,4 +1052,49 @@ describe('lib/metrics', function() {
         });
     });
   });
+
+  describe('logUserPreferences', () => {
+    const userPrefMetrics = {
+      'account-recovery': true,
+      'two-step-authentication': false,
+    };
+
+    it('correctly stores user preference', () => {
+      assert.deepEqual(metrics.getAllData().userPreferences, {});
+      metrics.logUserPreferences('account-recovery', true);
+      metrics.logUserPreferences('two-step-authentication', false);
+      assert.deepEqual(metrics.getAllData().userPreferences, userPrefMetrics);
+    });
+
+    it('correctly reports user preferences', () => {
+      sinon.stub(metrics, '_send').callsFake(() => Promise.resolve(true));
+      sinon.stub(metrics, '_isFlushRequired').callsFake(() => true);
+
+      return metrics
+        .flush()
+        .then(() => {
+          assert.deepEqual(metrics.getAllData().userPreferences, {});
+
+          metrics.logUserPreferences('account-recovery', true);
+          metrics.logUserPreferences('two-step-authentication', false);
+
+          return metrics.flush();
+        })
+        .then(() => {
+          // a second flush!
+          assert.deepEqual(
+            metrics._send.args[1][0].userPreferences,
+            userPrefMetrics
+          );
+
+          return metrics.flush();
```

Comment:
Why is this flush() necessary?

Reply:
Looks like I was following the convention in tests above, however this doesn't really need the second metrics flush, consider it removed!